### PR TITLE
sourcemap fix

### PR DESCRIPTION
**Pull request by STRd6**

### CODE REVIEW

File: integration/unplugin/src/index.ts [hunk 1]
```diff
@@ -204,7 +204,7 @@ const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
       sourceMaps.set(path.resolve(process.cwd(), id), compiled.sourceMap);
 
       const jsonSourceMap = compiled.sourceMap.json(
-        path.basename(id.replace(/\.tsx$/, '')),
+        path.basename(id.replace(/\.[t]sx$/, '')),
```
Comment (edemaine) [inline integration/unplugin/src/index.ts:207]: What is the purpose of this change? Is it supposed to be `[jt]`?

Comment (STRd6) [reply integration/unplugin/src/index.ts:207]: This fixes the sourcemap `sources` key from being `main.civet.jsx` and makes it the correct `main.civet` when building using esbuild.

Comment (STRd6) [reply integration/unplugin/src/index.ts:207]: > What is the purpose of this change? Is it supposed to be `[jt]sx`?

Oops, misread, yes it is supposed to be that. Now to figure out why it appeared to be working as is... :P

File: source/util.civet [hunk 2]
```diff
@@ -68,6 +68,8 @@ export SourceMap = (sourceString) ->
       mappings: @renderMappings()
       names: []
       sourcesContent: [sourceString]
+      toString: ->
+        JSON.stringify(this)
```
Comment (edemaine) [inline source/util.civet:72]: Should this be a method on the class or the `json` object? From the failing test, I assume the former, so this should be dedented one level.

**Review by edemaine — APPROVED** [review]
